Glossary: interpolation (#32877)
mdn/content: files/en-us/glossary/interpolation/index.md

{{GlossarySidebar}}

> Interpolation is a process of calculating values based on known values. The interpolation is used to obtain intermediate values of properties, such as height, width, etc., during the process of their animation.

Reviewer: The second sentence can't start with "The". This might be better.

```suggestion
Interpolation is a process for estimating unknown values in a range based on the values of known end points.
It is used, for example, to calculate intermediate values of HTML element properties when they are animated, such as height, width, and so on.
```

> Interpolation is a process of calculating values based on known values. The interpolation is used to obtain intermediate values of properties, such as height, width, etc., during the process of their animation.
> 
> Color interpolation is used to define intermediate values of colors with gradients, compositing, filters, transitions, animations, and color mixing and color modification functions.

Reviewer: ```suggestion
Interpolation is a process for estimating unknown values in a range, based on the values of known end points.

Interpolation used, for example, to calculate intermediate values of HTML element properties when they are animated, such as height, width, and so on.
Color interpolation is used to define intermediate values of colors with gradients, compositing, filters, transitions, animations, and color mixing and color modification functions.
```

{{GlossarySidebar}}

> Interpolation is a process for estimating unknown values in a range, based on the values of known end points.

Reviewer 2: Sorry I couldn't finish my review earlier, what do you think about this?

```suggestion
Interpolation is a method for estimating new data points based on a set of known data points.
```

> Interpolation used, for example, to calculate intermediate values of HTML element properties when they are animated, such as height, width, and so on.

Reviewer 2: ```suggestion
Interpolation is used, for example, to calculate intermediate values of HTML element properties when they are animated, such as height, width, and so on.
```

> Interpolation is used, for example, to calculate intermediate values of HTML element properties when they are animated, such as height, width, and so on.
> Color interpolation is used to define intermediate values of colors with gradients, compositing, filters, transitions, animations, and color mixing and color modification functions.

Reviewer 2: Minor reword for you to consider, feel free to dismiss if you prefer yours:

```suggestion
Interpolation calculates intermediate values in animations that change HTML properties such as height, width, etc.
Color interpolation defines intermediate values of colors in color mixing, gradients, compositing, filters, transitions, animations, and color functions.
```

> Interpolations aren't necessarily linear. When there is no mid-point between two values, such as values that accept only integers, the interpolation is discrete, with the value jumping from one value to the next without any interpolation between the discrete values. Most numeric values are interpolated as real, floating-point numbers. The progression of an interpolation depends on other properties, such [easing functions](/en-US/docs/Web/CSS/easing-function) with animations and color stops and {{CSSXref("color-interpolation-method", "color")}} and {{CSSXref("hue-interpolation-method", "hue")}} interpolation methods with gradients.

Reviewer 2: Another minor rejig of this, what do you think? Some small changes that might help connect the dots:

```suggestion
Interpolations aren't necessarily linear. Most interpolated values are real, floating-point numbers; however, when there is no mid-point between two values, such as values that accept only integers, the [interpolation is discrete](/en-US/docs/Web/CSS/integer#interpolation). The progression of interpolation also depends on [easing functions](/en-US/docs/Web/CSS/easing-function) in animations and color stops, and {{CSSXref("color-interpolation-method", "color")}} and {{CSSXref("hue-interpolation-method", "hue")}} interpolation methods in gradients.
```
